api/deployments/component_handler.go

```diff
@@ -2,435 +2,78 @@ package deployments
-			for _, envVariable := range container.Env {
-				if crdUtils.IsRadixEnvVar(envVariable.Name) {
-					radixEnvironmentVariables[envVariable.Name] = envVariable.Value
-				}
-			}
-		}
-	}
-
-	return radixEnvironmentVariables
-}
-
-func getReplicaSummaryList(pods []corev1.Pod, lastEventWarnings event.LastEventWarnings) []deploymentModels.ReplicaSummary {
-	return slice.Map(pods, func(pod corev1.Pod) deploymentModels.ReplicaSummary {
-		return deploymentModels.GetReplicaSummary(pod, lastEventWarnings[pod.GetName()])
-	})
-}
-
-func getAuxiliaryResources(ctx context.Context, kubeClient kubernetes.Interface, appName string, component v1.RadixCommonDeployComponent, envNamespace string) (auxResource deploymentModels.AuxiliaryResource, err error) {
-	if auth := component.GetAuthentication(); component.IsPublic() && auth != nil && auth.OAuth2 != nil {
-		auxResource.OAuth2, err = getOAuth2AuxiliaryResource(ctx, kubeClient, appName, component.GetName(), envNamespace)
-		if err != nil {
-			return
-		}
+	scaledObjects, err := kubequery.GetScaledObjectsForEnvironment(ctx, deploy.accounts.UserAccount.KedaClient, appName, envName)
+	if err != nil {
+		return nil, err
 	}
-
-	return
-}
-
-func getOAuth2AuxiliaryResource(ctx context.Context, kubeClient kubernetes.Interface, appName, componentName, envNamespace string) (*deploymentModels.OAuth2AuxiliaryResource, error) {
-	var oauth2Resource deploymentModels.OAuth2AuxiliaryResource
-	oauthDeployment, err := getAuxiliaryResourceDeployment(ctx, kubeClient, appName, componentName, envNamespace, defaults.OAuthProxyAuxiliaryComponentType)
+	noJobPayloadReq, err := labels.NewRequirement(kube.RadixSecretTypeLabel, selection.NotEquals, []string{string(kube.RadixSecretJobPayload)})
 	if err != nil {
 		return nil, err
 	}
-	if oauthDeployment != nil {
-		oauth2Resource.Deployment = *oauthDeployment
+	secretList, err := kubequery.GetSecretsForEnvironment(ctx, deploy.accounts.UserAccount.Client, appName, envName, *noJobPayloadReq)
+	if err != nil {
+		return nil, err
 	}
-
-	return &oauth2Resource, nil
-}
-
-func getAuxiliaryResourceDeployment(ctx context.Context, kubeClient kubernetes.Interface, appName, componentName, envNamespace, auxType string) (*deploymentModels.AuxiliaryResourceDeployment, error) {
-	var auxResourceDeployment deploymentModels.AuxiliaryResourceDeployment
-
-	selector := labelselector.ForAuxiliaryResource(appName, componentName, auxType).String()
-	deployments, err := kubeClient.AppsV1().Deployments(envNamespace).List(ctx, metav1.ListOptions{LabelSelector: selector})
+	eventList, err := kubequery.GetEventsForEnvironment(ctx, deploy.accounts.UserAccount.Client, appName, envName)
 	if err != nil {
 		return nil, err
 	}
-	if len(deployments.Items) == 0 {
-		auxResourceDeployment.Status = deploymentModels.ComponentReconciling.String()
-		return &auxResourceDeployment, nil
+	certs, err := kubequery.GetCertificatesForEnvironment(ctx, deploy.accounts.UserAccount.CertManagerClient, appName, envName)
```

Comment:
```suggestion
	certs, err := kubequery.GetCertificatesForEnvironment(ctx, deploy.accounts.ServiceAccount.CertManagerClient, appName, envName)
```
UserAccount does not have access (yet), so we must use ServiceAccount until fixed in radix-operator